plugin/vim-gh-line.vim

```diff
@@ -122,7 +122,7 @@ func! s:Action(origin, action)
 endfunc
 
 func! s:Commit(cdDir)
-  if exists('g:gh_use_canonical')
+  if exists('g:gh_use_canonical') && g:gh_use_canonical > 0
     return system(a:cdDir . 'git rev-parse HEAD')
```

Comment:
Can you please add a test for this. In [here](https://github.com/ruanyl/vim-gh-line/blob/b2d1cb78d5dfa9099b1402e0407f685f6800129b/test/vim-gh-line_test.vim#L18)

Roughly it should look like this: ( I am possibly not writing syntactically correct code)

```
func! s:testCommit(sid)
    call s:persistedPrint('Calling Commit')
    let l:fileDir = resolve(expand("%:p:h"))
    let l:cdDir = "cd '" . fileDir . "'; "
 
    let l:branch = system(l:cdDir . 'git rev-parse HEAD')
    
    let g:gh_use_canonical = 1
    let l:act = s:callWithSID(a:sid, 'Commit', l:cdDir)
    call assert_match(l:branch, l:act, 'Expected to find branch name in the Commit output ')
    unlet g:gh_use_canonical

endfunction 
```

Reply:
The new `s:testCommit(sid)` function should be called also from [here](https://github.com/ruanyl/vim-gh-line/blob/b2d1cb78d5dfa9099b1402e0407f685f6800129b/test/vim-gh-line_test.vim#L29)

Reply:
Added test case with one change instead of `git rev-parse HEAD` I've put `git rev-parse --abbrev-ref HEAD` since I think that test case was suppose to check for branch name based not canonical.